### ISSUE TITLE
Unable to initialize plugins via options

### ISSUE THREAD

**Issue by mmcc** [videojs/video.js#2510]
Looks like at some point we lost the ability to initialize a plugin via the `plugins` options object. Using the plugins example in sandbox:

### Works fine
```javascript
progressed = function(options) {
  this.on('progress', function(e) {
    console.log(this.bufferedPercent());
  });
};

// register the plugin
videojs.plugin('progressed', progressed);

// initialize it
vid1 = videojs('vid1');
vid1.progressed();
```

### Sadness
```javascript
var progressed = function(options) {
    this.on('progress', function(e) {
    console.log(this.bufferedPercent());
  });
};

videojs.plugin('progressed', progressed);

var vid1 = videojs('vid1', { plugins: { progressed: true }});
// => Uncaught TypeError: Cannot assign to read only property 'playerOptions' of true
```

**Comment by mmcc** [comment]
Assuming this is associated with #1980, but need to confirm.

**Comment by heff** [comment]
The issue is `progressed: true`. It expects an object for plugin options (not a boolean) and tries to add the player options to it. We probably broke it when we started passing playerOptions around.

**Comment by heff** [comment]
Closed by #2532
